Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -2097,18 +2103,21 @@ void CGContextSetPatternPhase(CGContextRef context, CGSize phase) {
                 positions[j] = DWRITE_GLYPH_OFFSET{ transformedPosition.x + run->glyphOffsets[j].advanceOffset,
                                                     std::round(transformedPosition.y + run->glyphOffsets[j].ascenderOffset) };
 
-                // Translate position of next glyph by current glyph's advance
-                runningPosition.x += run->glyphAdvances[j];
+                if ((run->bidiLevel & 1) == 0) {
+                    // Translate position of next glyph by current glyph's advance
+                    runningPosition.x += run->glyphAdvances[j];
+                }
             }
 
+            // Even if text is RTL set bidiLevel to 0 to draw in correct orientation for proper transformation by glyph
```

Comment:
nit: comment "we have compensated for the RTL glyph positions, so we don't need DWrite to do it for us."